chain-abci/src/app/commit.rs

```diff
@@ -43,7 +43,7 @@ impl<T: EnclaveProxy> ChainNodeApp<T> {
         let mut new_state = orig_state.expect("executing block commit, but no app state stored (i.e. no initchain or recovery was executed)");
         let mut resp = ResponseCommit::new();
         let mut inittx = self.storage.db.transaction();
-        if !self.delivered_txs.is_empty() {
```

Comment:
I'd suggest to keep this `!self.delivered_txs.is_empty()` check, but to move out the steps  on L123 to L142

Reply:
or remove the block scope as @calvinaco suggested https://github.com/crypto-com/chain/pull/530#discussion_r339448866

Reply:
ok, i'll modify